ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/services/commands/tracker/CompletionResponse.scala

```diff
@@ -77,10 +93,38 @@ object CompletionResponse {
     }
   }
 
+  private[daml] def toException(response: TrackedCompletionFailure): StatusException = {
+    response match {
+      case QueueCompletionFailure(failure) =>
+        toException(failure)
+      case QueueSubmitFailure(status) =>
+        val protoStatus =
+          rpc.Status
+            .newBuilder()
+            .setCode(status.getCode.value())
+            .setMessage(Option(status.getDescription).getOrElse("Failed to submit request"))
+        buildException(Map.empty[String, String], protoStatus)
+    }
+  }
+
   def toException(response: CompletionResponse.CompletionFailure): StatusException = {
-    val errorInfo = rpc.ErrorInfo.newBuilder().putAllMetadata(response.metadata.asJava).build()
+    val metadata = response.metadata
+    val status = extractStatus(response)
+    buildException(metadata, status)
+  }
+
+  private def buildException(metadata: Map[String, String], status: rpc.Status.Builder) = {
```

Comment:
This seems like a fairly generic gRPC utility, perhaps it can be pushed down into an utility library?

Reply:
Not sure where to put it tbh. I would extract it only if we identify another usage for it though.